### PR TITLE
Speed up some of the slowest pilot tests

### PR DESCRIPTION
**Pull request by howardjohn**
For https://github.com/istio/istio/issues/37555

### CODE REVIEW

File: pilot/pkg/model/network_test.go [hunk 1]
```diff
@@ -31,18 +31,19 @@ import (
 	"istio.io/istio/pilot/pkg/xds"
 	"istio.io/istio/pkg/config/mesh"
 	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/retry"
 )
 
 func TestGatewayHostnames(t *testing.T) {
 	origMinGatewayTTL := model.MinGatewayTTL
-	model.MinGatewayTTL = 3 * time.Second
+	model.MinGatewayTTL = 30 * time.Millisecond
```
Comment (howardjohn) [inline pilot/pkg/model/network_test.go:39]: drop TTL and instead do not assert we renew exactly once, just that we renew

File: pilot/pkg/model/network_test.go [hunk 2]
```diff
@@ -31,18 +31,19 @@ import (
...
 	t.Cleanup(func() {
 		model.MinGatewayTTL = origMinGatewayTTL
 	})
 
 	gwHost := "test.gw.istio.io"
-	dnsServer := newFakeDNSServer(":10053", 1, sets.NewSet(gwHost))
-	model.NetworkGatewayTestDNSServers = []string{"localhost:10053"}
+	dnsServer := newFakeDNSServer(":0", 1, sets.NewSet(gwHost))
```
Comment (howardjohn) [inline pilot/pkg/model/network_test.go:45]: Get random get port, to avoid conflict

File: pilot/pkg/model/network_test.go [hunk 3]
```diff
@@ -118,6 +124,7 @@ func newFakeDNSServer(addr string, ttl uint32, hosts sets.Set) *fakeDNSServer {
 			scopes.Framework.Errorf("fake dns server error: %v", err)
 		}
 	}()
+	waitLock.Lock()
```
Comment (howardjohn) [inline pilot/pkg/model/network_test.go:127]: wait for server to be ready. Copied from the miekg/dns tests

File: pilot/pkg/serviceregistry/kube/controller/endpointslice_test.go [hunk 4]
```diff
@@ -114,7 +114,7 @@ func TestEndpointSliceFromMCSShouldBeIgnored(t *testing.T) {
 	createEndpoints(t, controller, svcName, ns, portNames, svc1Ips, nil, map[string]string{
 		mcs.LabelServiceName: svcName,
 	})
-	if ev := fx.WaitForDuration("eds", 2*time.Second); ev != nil {
+	if ev := fx.WaitForDuration("eds", 200*time.Millisecond); ev != nil {
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/kube/controller/endpointslice_test.go:117]: No need to wait so long for negative case

File: pilot/pkg/trustbundle/trustbundle.go [hunk 5]
```diff
@@ -191,17 +191,17 @@ func (tb *TrustBundle) UpdateTrustAnchor(anchorConfig *TrustAnchorUpdate) error
 }
 
 func (tb *TrustBundle) updateRemoteEndpoint(spiffeEndpoints []string) {
-	tb.mutex.RLock()
+	tb.endpointMutex.RLock()
```
Comment (howardjohn) [inline pilot/pkg/trustbundle/trustbundle.go:194]: This is a real bug exposed by the test change

File: pilot/pkg/trustbundle/trustbundle_test.go [hunk 6]
```diff
@@ -262,7 +262,7 @@ func expectTbCount(t *testing.T, tb *TrustBundle, expAnchorCount int, ti time.Du
 			return fmt.Errorf("%s. Got %v, expected %v", strPrefix, len(certs), expAnchorCount)
 		}
 		return nil
-	}, retry.Timeout(ti), retry.Delay(100*time.Millisecond))
+	}, retry.Timeout(ti))
```
Comment (howardjohn) [inline pilot/pkg/trustbundle/trustbundle_test.go:265]: switch to backoff retry

File: pilot/pkg/trustbundle/trustbundle_test.go [hunk 7]
```diff
@@ -290,7 +290,7 @@ func TestAddMeshConfigUpdate(t *testing.T) {
 	tb := NewTrustBundle(caCertPool)
 
 	// Change global remote timeout interval for the duration of the unit test
-	remoteTimeout = 300 * time.Millisecond
+	remoteTimeout = 30 * time.Millisecond
```
Comment (howardjohn) [inline pilot/pkg/trustbundle/trustbundle_test.go:293]: speed things up

File: security/pkg/k8s/chiron/utils.go [hunk 8]
```diff
@@ -94,11 +94,9 @@ func GenKeyCertK8sCA(client clientset.Interface, dnsName,
 // 2. Approve a CSR
 // 3. Read the signed certificate
 // 4. Clean up the artifacts (e.g., delete CSR)
-func SignCSRK8s(client clientset.Interface,
-	csrData []byte, signerName string, requestedDuration *time.Duration,
-	usages []certv1.KeyUsage,
-	dnsName, caFilePath string,
-	approveCsr bool, appendCaCert bool, requestedLifetime time.Duration) ([]byte, []byte, error) {
+func SignCSRK8s(client clientset.Interface, csrData []byte, signerName string, usages []certv1.KeyUsage,
+	dnsName, caFilePath string, approveCsr, appendCaCert bool, requestedLifetime time.Duration,
```
Comment (ramaraochavali) [inline security/pkg/k8s/chiron/utils.go:98]: Is this new formatting standard? I also got in couple of my changes. Wondering what triggered this?

Comment (howardjohn) [reply security/pkg/k8s/chiron/utils.go:98]: seems kind of weird to me as well but I guess so. I think it only happens on really long functions

I don't really care personally as long as it's automated and consistent

Comment (ramaraochavali) [reply security/pkg/k8s/chiron/utils.go:98]: Yeah I also do not care. Was googling and found these https://dave.cheney.net/2014/10/04/that-trailing-comma and https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8.

Was wondering what change is triggering these now- but as you said, as long as we are consistent it is good.

**Comment by howardjohn** [comment]
need to fix one

**Review by howardjohn — COMMENTED** [review]
Broken test is fixed in https://github.com/istio/istio/pull/37839

**Review by nmittler — APPROVED** [review]

**Review by douglas-reid — APPROVED** [review]
approving e&T files.

**Review by lei-tang — APPROVED** [review]

**Review by stevenctl — APPROVED** [review]

**Pull request merged**